### PR TITLE
Exposed Coordinates to Dynamo 2016

### PR DESCRIPTION
**Pull request by moethu**
### Purpose

Exposed Revit Survey and Base point to Dynamo

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] Snapshot of UI changes, if any.

### Reviewers

@mjkkirschner 

### FYIs
@sharadkjaiswal

### CODE REVIEW

File: src/Libraries/RevitNodes/Elements/Coordinates.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+using Dynamo.Models;
+using Revit.GeometryConversion;
+
+namespace Revit.Elements
+{
+
+    /// <summary>
+    /// Rebar Nodes for Coordinates
+    /// </summary>
+    public static class Coordinates
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/Coordinates.cs]: can you expand the summary a bit for this class?

File: src/Libraries/RevitNodes/Elements/Coordinates.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    {
+        /// <summary>
+        /// Get Base or SurveyPoint
+        /// </summary>
+        /// <param name="surveypoint"></param>
+        /// <returns></returns>
+        [IsVisibleInDynamoLibrary(false)]
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/Coordinates.cs]: I don't think you need this attribute on a private method.

File: src/Libraries/RevitNodes/Elements/Coordinates.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            RVT.BuiltInCategory category = (surveypoint)? RVT.BuiltInCategory.OST_SharedBasePoint : RVT.BuiltInCategory.OST_ProjectBasePoint;
+
+            // Get Revit document
+            RVT.Document doc = RevitServices.Persistence.DocumentManager.Instance.CurrentDBDocument;
+            
+            // Get all elements of the previously selected category
+            Autodesk.Revit.DB.FilteredElementCollector collector = new RVT.FilteredElementCollector(doc).OfCategory(category);
+            
+            // Get the first element (should only be one)
+            RVT.BasePoint element = (RVT.BasePoint)collector.ToElements().FirstOrDefault();
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/Coordinates.cs]: you should check the element is not null.

File: src/Libraries/RevitNodes/Elements/Coordinates.cs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        /// <summary>
+        /// Get Project Rotation
+        /// </summary>
+        /// <returns>Rotation in degrees</returns>
+        public static double ProjectRotation()
+        {
+            // Get project base point
+            RVT.Document doc = RevitServices.Persistence.DocumentManager.Instance.CurrentDBDocument;
+            Autodesk.Revit.DB.FilteredElementCollector collector = new RVT.FilteredElementCollector(doc).OfCategory(RVT.BuiltInCategory.OST_ProjectBasePoint);            
+            RVT.BasePoint element = (RVT.BasePoint)collector.ToElements().FirstOrDefault();
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/Coordinates.cs]: same null check

File: src/Libraries/RevitNodes/Elements/Coordinates.cs [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+            // Return the rotation parameter as double
+            RVT.Parameter param = element.get_Parameter(RVT.BuiltInParameter.BASEPOINT_ANGLETON_PARAM);
+            return param.AsDouble();
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodes/Elements/Coordinates.cs]: hmm, is this AsDouble() call effected by units settings in Revit?

**Comment by mjkkirschner** [comment]
@moethu this looks pretty good,  a few questions, and please add 1 or 2 tests covering these methods.

**Comment by moethu** [comment]
@mjkkirschner applied all requested changes.

**Comment by mjkkirschner** [comment]
still merge conflicts here

**Comment by moethu** [comment]
@mjkkirschner resolved

**Comment by mjkkirschner** [comment]
please create the other revit branches.

**Comment by mjkkirschner** [comment]
@moethu after merging some of the other PRs these PRS (coordinates) have merge conflicts.

**Comment by moethu** [comment]
@mjkkirschner should be fine now

File: test/Libraries/RevitNodesTests/Elements/CoordinatesTests.cs [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+            Assert.IsTrue(basepoint.Equals(Autodesk.DesignScript.Geometry.Point.Origin()));
+        }
+
+        [Test]
+        [TestModel(@".\Empty.rvt")]
+        public void SurveyPoint_Valid()
+        {
+            var spoint = Coordinates.SurveyPoint();
+            Assert.IsNotNull(spoint);
+            Assert.IsTrue(spoint.Equals(Autodesk.DesignScript.Geometry.Point.Origin()));
```
Comment (mjkkirschner) [inline test/Libraries/RevitNodesTests/Elements/CoordinatesTests.cs]: does this actually pass? I think you may need to use IsAlmostEqual?

**Comment by mjkkirschner** [comment]
@moethu I am going to merge this series of PRS (coordinates) but lets get all tests passing before moving on to any other merges. I believe this PR will introduce one failing test.

**Pull request merged**